src/snowflake/snowpark/modin/plugin/_internal/apply_utils.py

```diff
@@ -221,7 +221,10 @@ def end_partition(self, df):  # type: ignore[no-untyped-def] # pragma: no cover
                 raise TypeError(f"Unsupported data type {df} from df.apply")
 
             result["value"] = (
-                result["value"].apply(handle_missing_value_in_variant).astype(object)
+                result["value"]
+                .apply(handle_missing_value_in_variant)
+                .apply(convert_numpy_int_result_to_int)
```

Comment:
Should we combine these into a single .apply call?

Reply:
I doubt it matters at this point given the huge overheads from temp table creation, joins, and UDTF invocation but it's better practice to combine the calls. Done.